File: 14-longest-common-prefix/14-longest-common-prefix.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        result = ""
        str1 = strs[0]
        for i in range(len(str1)):
            for j in strs:
                if i == len(j) or j[i] != strs[0][i]:
                    return result
            result += strs[0][i]
        return result
                
        

File: 14-longest-common-prefix/14-longest-common-prefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        string ans;
        int n = strs[0].length();
        for(int i = 0; i < n; i++){
            char ch = strs[0][i];
            int j;
            for(j = 1; j < strs.size(); j++){
                if (strs[j][i] != ch)
                    break;
            }
        if (j == strs.size()){
            ans += strs[0][i];
        }
        else{
            //cout << ans;
            return ans;
        }
    }
        
    return ans;
    }
};